vividus-agent-reportportal/src/main/java/org/vividus/reportportal/config/condition/ReportPortalEnableCondition.java

```diff
@@ -18,16 +18,10 @@
 
 import com.epam.reportportal.utils.properties.ListenerProperty;
 
-import org.springframework.context.annotation.Condition;
-import org.springframework.context.annotation.ConditionContext;
-import org.springframework.core.type.AnnotatedTypeMetadata;
-
-public class ReportPortalEnableCondition implements Condition
+public class ReportPortalEnableCondition extends AbstractBooleanCondition
 {
-    @Override
-    public boolean matches(ConditionContext context, AnnotatedTypeMetadata metadata)
+    public ReportPortalEnableCondition()
     {
-        String enableProperty = ListenerProperty.ENABLE.getPropertyName();
-        return context.getEnvironment().getProperty(enableProperty, boolean.class);
```

Comment:
as per javadoc "Return the property value associated with the given key, or null if the key cannot be resolved." I think it might lead to unexpected result in case of missing property

Reply:
I think it is not the case, because we have the default value